### PR TITLE
[NVIDIA XLA GPU] Expose collective matmul trigger in debug options

### PR DESCRIPTION
**Pull request by Tixxx**
expose threshold_for_windowed_einsum_mib through debug options as the first step of enabling collective matmul for gpu.

### CODE REVIEW

**Review by akuegel — APPROVED** [review]

**Comment by gbaned** [comment]
Hi @Tixxx Can you please rebase your branch and resolve the conflicts? Thank you!

**Comment by Tixxx** [comment]
> Hi @Tixxx Can you please rebase your branch and resolve the conflicts? Thank you!

Pushed a rebase. Can we merge this soon to avoid further conflicts?

**Comment by gbaned** [comment]
Hi @akuegel / @frgossen Can you please review this PR ? Thank you!

**Review by akuegel — APPROVED** [review]

**Review by frgossen — APPROVED** [review]

File: xla/debug_options_flags.cc [hunk 1]
```diff
@@ -1428,6 +1428,13 @@ void MakeDebugOptionsFlags(std::vector<tsl::Flag>* flag_list,
       bool_setter_for(&DebugOptions::set_xla_gpu_enable_cub_radix_sort),
       debug_options->xla_gpu_enable_cub_radix_sort(),
       "Enable radix sort using CUB for simple shapes"));
+  flag_list->push_back(tsl::Flag(
+      "xla_gpu_threshold_for_windowed_einsum_mib",
+      int64_setter_for(
+          &DebugOptions::set_xla_gpu_threshold_for_windowed_einsum_mib),
+      debug_options->xla_gpu_threshold_for_windowed_einsum_mib(),
+      "Threshold to enable windowed einsum (collective matmul) in MB."
+      "Default is 100000"));
```
Comment (nouiz) [inline xla/debug_options_flags.cc:1437]: I always find that kind of comment unclear.
Is collective_matmul enabled when one input tensor is above that size? When the sum of both is above that size? Under that size? Can you clarify this here?

Comment (Tixxx) [reply xla/debug_options_flags.cc:1437]: sorry i was OOO for the last 2 weeks and missed this comment. This is actually for just lhs or rhs of the dot depending on the sharding strategy. I will make this more descriptive in the next pr.

**Pull request closed without merging**